client/client-multi/client-cpp/impl/channel/impl/DefaultOperationTcpChannel.cpp

```diff
@@ -252,6 +252,10 @@ void DefaultOperationTcpChannel::openConnection()
         return;
     }
 
+    /* Check whether ip has been changed */
+    context_.getClientStateListener()->onConnectionEstablished(sock_->local_endpoint().address().to_string(),
```

Comment:
I think it is better to notify the channel manager about new connection and the latter will pass this info to the listener or to some other modules. In other words to use the same approach like in the case when connection is failed.
Besides I suggest to wrap both parameters in a single structure like IPConnectionInfo. Using that approach lets us simply add a field or two if we want to specify destination's host/port.